lib/powerline.sh

```diff
@@ -101,7 +101,7 @@ __process_powerline() {
 		local separator_background_color=${powerline_segment[4]}
 		local separator_foreground_color=${powerline_segment[5]}
 
-		eval "__print_${side}_segment ${segment_index} ${background_color} ${foreground_color} ${separator} ${separator_background_color} ${separator_foreground_color}"
+		eval "__print_${side}_segment \"${segment_index}\" \"${background_color}\" \"${foreground_color}\" \"${separator}\" \"${separator_background_color}\" \"${separator_foreground_color}\""
```

Comment:
Because `eval`.
Are you asking why it's double quotes as opposed to single quotes?
Because in general, each segment has to be passed to `__print_*_segment` as a single argument, so it needs to be enclosed in quotes.